### PR TITLE
Implement B019

### PR DESCRIPTION
**Pull request by harupy**
#389

### CODE REVIEW

File: src/flake8_bugbear/plugins/cached_instance_method.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+use crate::ast::types::{Range, ScopeKind};
+use crate::check_ast::Checker;
+use crate::checks::{Check, CheckKind};
+
+const CACHE_FUNCTIONS: [&str; 4] = [
+    "functools.lru_cache",
+    "functools.cache",
+    "lru_cache",
+    "cache",
+];
```
Comment (charliermarsh) [inline src/flake8_bugbear/plugins/cached_instance_method.rs:13]: Do you mind tweaking this to use the `checker.from_imports` to verify that these are actually imported from `functools`?

Comment (charliermarsh) [reply src/flake8_bugbear/plugins/cached_instance_method.rs:13]: (There are some TODOs to do this in other files too, like `src/flake8_bugbear/plugins/mutable_argument_default.rs`, but it's a newer feature.)

Comment (harupy) [reply src/flake8_bugbear/plugins/cached_instance_method.rs:13]: Got it. Do we have an example?

Comment (charliermarsh) [reply src/flake8_bugbear/plugins/cached_instance_method.rs:13]: `flake8_2020/plugins.rs`

Comment (harupy) [reply src/flake8_bugbear/plugins/cached_instance_method.rs]: Done!

**Pull request merged**